### PR TITLE
Gen8avc

### PR DESCRIPTION
**Pull request by laureatian**

### CODE REVIEW

**Comment by xhaihao** [comment]
Each patch shouldn't break compilation. could you please squash the last 4 commits into the corresponding commit?

File: src/gen9_avc_encoder.c [hunk 1]
```diff
@@ -860,7 +992,7 @@ gen9_avc_allocate_resources(VADriverContextP ctx,
         goto failed_allocation;
 
     /* scaling related surface   */
-    if (avc_state->mb_status_supported) {
+    if (!IS_GEN8(i965->intel.device_info)) {
```
Comment (QuPengfei) [inline src/gen9_avc_encoder.c]: it looks confused and "if (avc_state->mb_status_supported)"  looks better. would you initialize the  variable mb_status_supported according to the GEN8 platform

File: src/gen9_avc_encoder.c [hunk 2]
```diff
@@ -5634,15 +5913,1028 @@ gen9_avc_kernel_sfd(VADriverContextP ctx,
...
+        curbe_cmd->dw10.mbv_proc_states_bti_top_field  = GEN8_SCALING_FIELD_TOP_MBVPROCSTATS_DST_CM;
+        curbe_cmd->dw11.mbv_proc_states_bti_bottom_field = GEN8_SCALING_FIELD_BOT_MBVPROCSTATS_DST_CM;
+    }
+
+    i965_gpe_context_unmap_curbe(gpe_context);
+    return;
+}
+
+static void
+gen8_avc_send_surface_scaling(VADriverContextP ctx,
```
Comment (QuPengfei) [inline src/gen9_avc_encoder.c]: it is almost the same as gen9_avc_send_surface_scaling. is it possible to reuse it?

File: src/gen9_avc_encoder.c [hunk 3]
```diff
@@ -5634,15 +5913,1028 @@ gen9_avc_kernel_sfd(VADriverContextP ctx,
...
+    cmd->dw85.brc_curbe_surf_index = GEN8_AVC_MBENC_BRC_CURBE_DATA_CM;
+    cmd->dw86.static_detection_cost_table_index = GEN8_AVC_MBENC_STATIC_FRAME_DETECTION_OUTPUT_CM;
+
+    i965_gpe_context_unmap_curbe(gpe_context);
+
+    return;
+}
+
+static void
+gen8_avc_send_surface_mbenc(VADriverContextP ctx,
```
Comment (QuPengfei) [inline src/gen9_avc_encoder.c]: it is almost the same as gen9_avc_send_surface_mbenc. is it possible to reuse it?

File: src/gen9_avc_encoder.c [hunk 4]
```diff
@@ -1867,6 +2091,48 @@ void gen9_avc_set_image_state(VADriverContextP ctx,
     return;
 }
 
+void gen8_avc_set_image_state(VADriverContextP ctx,
```
Comment (QuPengfei) [inline src/gen9_avc_encoder.c]: is it almost same as gen9_avc_set_image_state. is it possible to reuse it?

**Pull request closed without merging**